packages/frint-router-react/src/Link.js

```diff
@@ -11,29 +13,49 @@ export default class Link extends React.Component {
 
     this.state = {
       active: false,
-    }
+    };
+
+    this.subscription = null;
   }
 
   componentDidMount() {
-    this.subscription = null;
+    this.considerSubscribingToRouter(this.props);
+  }
 
-    if (typeof this.props.activeClassName === 'string') {
-      this.subscription = this.context.app
-        .get('router')
-        .getMatch$(this.props.to, {
-          exact: this.props.exact ? true : false,
-        })
-        .subscribe((matched) => {
-          if (!matched) {
-            return this.setState({ active: false });
-          }
-
-          return this.setState({ active: true });
-        });
-    }
+  componentWillReceiveProps(nextProps) {
+    this.considerSubscribingToRouter(nextProps);
   }
 
   componentWillUnmount() {
+    this.unsubscribeFromRouter();
+  }
+
+  considerSubscribingToRouter(nextProps) {
+    if (typeof nextProps.activeClassName === 'string') {
```

Comment:
consider returning early, to avoid nesting:

```js
if (typeof nextProps.activeClassName !== 'string') {
  return;
}

// ... rest of the code
```